packages/js/components/src/experimental-select-tree-control/select-tree.tsx

```diff
@@ -268,6 +297,46 @@ export const SelectTree = function SelectTree( {
 											props.onRemove( item );
 										}
 									} }
+									onBlur={ ( event ) => {
+										if ( isEventOutside( event ) ) {
+											setIsOpen( false );
+											setIsFocused( false );
+										}
+									} }
+									onKeyDown={ ( event ) => {
+										if (
+											event.key === 'ArrowLeft' ||
+											event.key === 'ArrowRight'
+										) {
+											const selectedItem = (
+												event.target as HTMLElement
+											 ).closest(
+												'.woocommerce-experimental-select-control__selected-item'
+											);
+											const sibling =
+												event.key === 'ArrowLeft'
+													? selectedItem?.previousSibling
+													: selectedItem?.nextSibling;
+											if ( sibling ) {
+												(
+													(
+														sibling as HTMLElement
+													 ).querySelector(
+														'.woocommerce-tag__remove'
+													) as HTMLElement
+												 )?.focus();
```

Comment:
This is hardcoding knowledge of the tag implementation. Seems like it would be better to just tell the tag to focus itself.

Reply:
I've refactored it using refs, please check it out. I don't see a easy way of telling the tag to focus itself, as it is a child of a child. I think using refs couple them a little less. Let me know if you think it's reasonable.